grp/basic.gd

```diff
@@ -343,64 +343,122 @@ end );
 
 #############################################################################
 ##
-#O  QuaternionGroupCons( <filter>, <n> )
+#O  DicyclicGroupCons( <filter>, <n> )
 ##
 ##  <ManSection>
-##  <Oper Name="QuaternionGroupCons" Arg='filter, n'/>
+##  <Oper Name="DicyclicGroupCons" Arg='filter, n'/>
 ##
 ##  <Description>
 ##  </Description>
 ##  </ManSection>
 ##
-DeclareConstructor( "QuaternionGroupCons", [ IsGroup, IsInt ] );
+DeclareConstructor( "DicyclicGroupCons", [ IsGroup, IsInt ] );
+DeclareSynonym("QuaternionGroupCons", DicyclicGroupCons);
 
 
 #############################################################################
 ##
-#F  QuaternionGroup( [<filt>, ]<n> )  . . . . . . . quaternion group of order <n>
+#F  DicyclicGroup( [<filt>, [<field>, ] ]<n> )  Dicyclic group of order <n>
 ##
-##  <#GAPDoc Label="QuaternionGroup">
+##  <#GAPDoc Label="DicyclicGroup">
 ##  <ManSection>
-##  <Func Name="QuaternionGroup" Arg='[filt, ]n'/>
-##  <Func Name="DicyclicGroup" Arg='[filt, ]n'/>
+##  <Func Name="DicyclicGroup" Arg='[filt, [field, ]]n'/>
+##  <Func Name="QuaternionGroup" Arg='[filt, [field, ]]n'/>
 ##
 ##  <Description>
-##  constructs the generalized quaternion group (or dicyclic group) of size
-##  <A>n</A> in the category given by the filter <A>filt</A>.  Here, <A>n</A>
-##  is a multiple of 4.
+##  constructs the dicyclic group of size <A>n</A> in the category given by the
+##  filter <A>filt</A>. Here, <A>n</A> is a multiple of 4.
```

Comment:
This text seems to refer to implicitly refer to `DicyclicGroup` only, but is of course listed directly after `QuaternionGroup`. I'd rephrase to e.g. "DicyclicGroup constucts ... " and then perhaps insert one sentence that mentions `QuaternionGroup` does the same but is restricted to inputs which are powers of 2, and that is provided for backwards compatibility.